pkg/controllers/monitoring-stack/controller.go

```diff
@@ -130,20 +131,74 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	availableCondition := metav1.Condition{
+		Type:               "Available",
+		ObservedGeneration: ms.ObjectMeta.Generation,
+		LastTransitionTime: metav1.Time{Time: time.Now().UTC()},
+	}
+
 	reconcilers := stackComponentReconcilers(ms, r.instanceSelectorKey, r.instanceSelectorValue)
 	for _, reconciler := range reconcilers {
 		err := reconciler(ctx, r.k8sClient, r.scheme)
 		// handle create / update errors that can happen due to a stale cache by
 		// retrying after some time.
 		if errors.IsAlreadyExists(err) || errors.IsConflict(err) {
 			logger.V(3).Info("skipping reconcile error", "err", err)
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{}, err
 		}
 	}
-	return ctrl.Result{}, nil
+	availableCondition.Status = metav1.ConditionTrue
+	availableCondition.Reason = "Ready"
+	availableCondition.Message = "Monitoring stack is up and running"
+	err = r.updateStatusCondition(ctx, ms, availableCondition)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+}
+
+func (r *reconciler) updateStatusCondition(ctx context.Context, ms *stack.MonitoringStack, c metav1.Condition) error {
+	updatedMS := ms.DeepCopy()
+	newStatus := stack.MonitoringStackStatus{}
+
+	// if the condition exists and the reason hasn't changed then do no update
+	if existingC := getConditionByType(c.Type, updatedMS.Status.Conditions); existingC != nil {
+		if existingC.Reason == c.Reason {
+			return nil
+		}
+	}
+	updatedMS.Status.Conditions = append(newStatus.Conditions, c)
+	err := r.k8sClient.Status().Update(ctx, updatedMS)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getConditionByType(ctype string, conditions []metav1.Condition) *metav1.Condition {
```

Comment:
```suggestion
func conditionForType(conditions []metav1.Condition, x conditionType) *metav1.Condition {
```
I am inclined to create an enum for the types e.g .. 
```
type conditionType string
const available conditionType = "Availble"
```

Reply:
Yes why not.... I was thinking about adding something like `map[conditionType]Condition` to the `reconciler` type, but not sure if it's necessary right now.

Reply:
Creating a type to indicate condition type is necessary even if there is only one condition to avoid any typos and to have a single place to change / make corrections.